src/Middleware/Diagnostics/src/DiagnosticsLoggerExtensions.cs

```diff
@@ -20,7 +21,9 @@ internal static class DiagnosticsLoggerExtensions
             LoggerMessage.Define(LogLevel.Error, new EventId(3, "Exception"), "An exception was thrown attempting to execute the error handler.");
 
         private static readonly Action<ILogger, Exception?> _errorHandlerNotFound =
-            LoggerMessage.Define(LogLevel.Warning, new EventId(4, "HandlerNotFound"), "No exception handler was found, rethrowing original exception.");
+            LoggerMessage.Define(LogLevel.Warning, new EventId(4, "HandlerNotFound"), $"The exception handler configured on ExceptionHandlerOptions produced a 404 status response. " +
+                $"An InvalidOperationException containing the original exception will be thrown since this is often due to a misconfigured ExceptionHandlerOptions.ExceptionHandlingPath. " +
+                $"If the exception handler is expected to return 404 status responses then set ExceptionHandlerOptions.AllowStatusCode404Response to true.");
```

Comment:
```suggestion
            LoggerMessage.Define(LogLevel.Warning, new EventId(4, "HandlerNotFound"), "The exception handler configured on ExceptionHandlerOptions produced a 404 status response. " +
                "An InvalidOperationException containing the original exception will be thrown since this is often due to a misconfigured ExceptionHandlerOptions.ExceptionHandlingPath. " +
                "If the exception handler is expected to return 404 status responses then set ExceptionHandlerOptions.AllowStatusCode404Response to true.");
```